### PR TITLE
fix: Convert CLSClustering and ThuNews to fast

### LINKED ISSUE #755: Convert ThuNews and CLS to Fast
Part of #660 .

### PR DESCRIPTION
**Pull request by isaac-chung**
## Checklist for adding MMTEB dataset

Reason for dataset addition: resolves #755 

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/ThuNewsClusteringFastS2S.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        0.2860376680688822,
+        0.30687457686544983,
+        0.28716007837426494,
+        0.31225412878214914,
+        0.3217343285029656,
+        0.3278291758899311
+      ]
+    }
+  }
+}
```
Comment (Ruqyai) [inline results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/ThuNewsClusteringFastS2S.json:23]: Great Job.. Thanks

File: docs/mmteb/points/757.jsonl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "isaac-chung", "Bug fixes": 8}
+{"GitHub": "???", "Review PR": 2}
```
Comment (Ruqyai) [inline docs/mmteb/points/757.jsonl:2]: Great work.. Don't forget to add the names of the reviewers

**Review by Ruqyai — APPROVED** [review]

File: mteb/tasks/Clustering/zho/CMTEBClustering.py [hunk 3]
```diff
@@ -1,11 +1,124 @@
...
+from mteb.abstasks.AbsTaskClusteringFast import AbsTaskClusteringFast
 from mteb.abstasks.TaskMetadata import TaskMetadata
 
-from ....abstasks.AbsTaskClustering import AbsTaskClustering
+NUM_SAMPLES = 2048
+
+
+class CLSClusteringFastS2S(AbsTaskClusteringFast):
+    metadata = TaskMetadata(
+        name="CLSClusteringFastS2S",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/zho/CMTEBClustering.py:16]:
```suggestion
        name="CLSClusteringS2S.v2",
```
To keep it consistent with other changes.

File: mteb/tasks/Clustering/zho/CMTEBClustering.py [hunk 4]
```diff
@@ -1,11 +1,124 @@
...
+            self.seed,
+            self.metadata.eval_splits,
+            label="labels",
+            n_samples=NUM_SAMPLES,
+        )
+
+
+class CLSClusteringFastP2P(AbsTaskClusteringFast):
+    metadata = TaskMetadata(
+        name="CLSClusteringFastP2P",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/zho/CMTEBClustering.py:69]:
```suggestion
        name="CLSClusteringP2P.v2",
```

File: mteb/tasks/Clustering/zho/CMTEBClustering.py [hunk 5]
```diff
@@ -1,11 +1,124 @@
...
+import itertools
+
+from datasets import Dataset, DatasetDict
+
+from mteb.abstasks.AbsTaskClustering import AbsTaskClustering
+from mteb.abstasks.AbsTaskClusteringFast import AbsTaskClusteringFast
 from mteb.abstasks.TaskMetadata import TaskMetadata
 
-from ....abstasks.AbsTaskClustering import AbsTaskClustering
+NUM_SAMPLES = 2048
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/zho/CMTEBClustering.py:11]: How well does the score compare? I could imagine that we might need more than 2048 (but I might be wrong).

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good only a few minor thinks! Good to have you onboard @Ruqyai

File: mteb/tasks/Clustering/zho/CMTEBClustering.py (hunk 3 above)
Comment (isaac-chung) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:16]: I just realized that `v2` was not added to my other conversions. Should we add that in https://github.com/embeddings-benchmark/mteb/issues/660? cc @x-tabdeveloping

File: mteb/tasks/Clustering/zho/CMTEBClustering.py (hunk 5 above)
Comment (isaac-chung) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:11]: Will run the full thing as well to compare.

File: mteb/tasks/Clustering/zho/CMTEBClustering.py (hunk 3 above)
Comment (x-tabdeveloping) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:16]: Mine either actually (though I converted most things to Hierarchical). I think we should prefer versioning over overwriting, but what's merged is merged

Comment (isaac-chung) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:16]: I'm confused about what is being overwritten here. And it's no problem - as we want to be consistent - they can be changed to align with others. 

If we want v2 then the guide needs updating as well. I personally have no preference. Could @KennethEnevoldsen @x-tabdeveloping you guys decide pleeeease?

File: mteb/tasks/Clustering/zho/CMTEBClustering.py (hunk 5 above)
Comment (isaac-chung) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:11]: - CLS P2P Fast: "evaluation_time": 5.24, "v_measure": 0.38429563913648135,
- CLS P2P: "evaluation_time": 156.89, "v_measure": 0.3919956383426169,

File: mteb/tasks/Clustering/zho/CMTEBClustering.py (hunk 3 above)
Comment (x-tabdeveloping) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:16]: I prefer the `.v2` personally, I think we should roll with that.

Comment (isaac-chung) [reply mteb/tasks/Clustering/zho/CMTEBClustering.py:16]: Thanks, I'll make the changes 👍

**Pull request merged**